experimental/handoff-library/csharp_dotnetcore/Microsoft.Bot.Builder.Handoff.Test/Mocks/MockHttpMessageHandler.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+using Microsoft.Bot.Schema.Handoff;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Rest;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.Handoff.UnitTests
+{
+    internal class MockHttpMessageHandler : HttpMessageHandler
```

Comment:
Consider Moq. Though Moq can get complicated. And sometimes its best to combine it with explicitly coded classes. Here are some examples:

https://github.com/microsoft/botbuilder-dotnet/blob/master/tests/integration/Microsoft.Bot.Builder.Integration.AspNet.Core.Tests/BotFrameworkHttpAdapterTests.cs